controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -500,9 +500,25 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
 	if err != nil {
 		return err
 	}
-	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil &&
-		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository != "" {
-		newImageName, err = containerutil.ModifyImageRepository(newImageName, kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository)
+	var imageRepository string
```

Comment:
It seems that the code that computes imageRepository is repeated in two places, what about moving to a func and re-use?

Reply:
Added a func `ImageRepositoryFromClusterConfig` which gets used in all 3 places.